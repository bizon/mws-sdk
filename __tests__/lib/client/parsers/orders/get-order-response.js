const parseXml = require('../../../../../lib/client/parsers')
const parseGetOrderResponse = require('../../../../../lib/client/parsers/orders/get-order-response')

describe('lib.client.parsers.orders.get-order-response', () => {
  it('should parse the GetOrderResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetOrderResponse xmlns="https://mws.amazonservices.com/Orders/2013-09-01">
        <GetOrderResult>
          <Orders>
            <Order>
              <AmazonOrderId>902-3159896-1390916</AmazonOrderId>
              <PurchaseDate>2017-01-20T19:49:35Z</PurchaseDate>
              <LastUpdateDate>2017-01-20T19:49:35Z</LastUpdateDate>
              <OrderStatus>Pending</OrderStatus>
              <FulfillmentChannel>MFN</FulfillmentChannel>
              <NumberOfItemsShipped>0</NumberOfItemsShipped>
              <NumberOfItemsUnshipped>0</NumberOfItemsUnshipped>
              <PaymentMethod>Other</PaymentMethod>
              <PaymentMethodDetails>
                <PaymentMethodDetail>CreditCard</PaymentMethodDetail>
                <PaymentMethodDetail>GiftCerificate</PaymentMethodDetail>
              </PaymentMethodDetails>
              <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>>
              <ShipmentServiceLevelCategory>Standard
              </ShipmentServiceLevelCategory>
              <OrderType>StandardOrder</OrderType>
              <EarliestShipDate>2017-01-20T19:51:16Z</EarliestShipDate>
              <LatestShipDate>2017-01-25T19:49:35Z</LatestShipDate>
              <IsBusinessOrder>false</IsBusinessOrder>
              <IsPrime>false</IsPrime>
              <IsPremiumOrder>false</IsPremiumOrder>
              <IsGlobalExpressEnabled>false</IsGlobalExpressEnabled>
            </Order>
          </Orders>
        </GetOrderResult>
        <ResponseMetadata>
          <RequestId>88faca76-b600-46d2-b53c-0c8c4533e43a</RequestId>
        </ResponseMetadata>
      </GetOrderResponse>`,
    )

    const response = parseGetOrderResponse('/orders:GetOrderResponse', doc)

    expect(response).toMatchSnapshot()
  })
})
