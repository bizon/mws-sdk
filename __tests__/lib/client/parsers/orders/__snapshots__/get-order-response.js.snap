// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.orders.get-order-response should parse the GetOrderResponse example response from MWS doc 1`] = `
Object {
  "getOrderResult": Object {
    "orders": Array [
      Object {
        "amazonOrderId": "902-3159896-1390916",
        "automatedShippingSettings": null,
        "buyerCounty": null,
        "buyerEmail": null,
        "buyerName": null,
        "buyerTaxInfo": null,
        "defaultShipFromLocationAddress": null,
        "earliestDeliveryDate": null,
        "earliestShipDate": 2017-01-20T19:51:16.000Z,
        "easyShipShipmentStatus": null,
        "fulfillmentChannel": "MFN",
        "isBusinessOrder": false,
        "isEstimatedShipDateSet": null,
        "isGlobalExpressEnabled": false,
        "isPremiumOrder": false,
        "isPrime": false,
        "isReplacementOrder": null,
        "isSoldByAB": null,
        "lastUpdateDate": 2017-01-20T19:49:35.000Z,
        "latestDeliveryDate": null,
        "latestShipDate": 2017-01-25T19:49:35.000Z,
        "marketplaceId": "ATVPDKIKX0DER",
        "numberOfItemsShipped": 0,
        "numberOfItemsUnshipped": 0,
        "orderChannel": null,
        "orderStatus": "Pending",
        "orderTotal": null,
        "orderType": "StandardOrder",
        "paymentExecutionDetail": Array [],
        "paymentMethod": "Other",
        "paymentMethodDetails": Array [
          "CreditCard",
          "GiftCerificate",
        ],
        "promiseResponseDueDate": null,
        "purchaseDate": 2017-01-20T19:49:35.000Z,
        "purchaseOrderNumber": null,
        "replacedOrderId": null,
        "salesChannel": null,
        "sellerOrderId": null,
        "shipServiceLevel": null,
        "shipmentServiceLevelCategory": "Standard",
        "shippingAddress": null,
      },
    ],
  },
  "responseMetadata": Object {
    "requestId": "88faca76-b600-46d2-b53c-0c8c4533e43a",
  },
}
`;
